### PR TITLE
LSO: use PageEditor for intro/extro

### PR DESCRIPTION
**Pull request by nhaagen**
This removes RTE from LSOs' settings and adds PageEditorComponents for Startbutton and Curriculum.
Please see the wiki for details: https://docu.ilias.de/goto_docu_wiki_wpage_6760_1357.html

### CODE REVIEW

**Comment by catenglaender** [comment]
I found a potential bug in the release_7 version of this PR that probably also occurs in the trunk version: Content Styles aren't applied on the LSO intro / extro pages - https://mantis.ilias.de/view.php?id=32543

File: Services/COPage/classes/class.ilPageEditorGUI.php [hunk 1]
```diff
@@ -31,6 +31,7 @@
  * @ilCtrl_Calls ilPageEditorGUI: ilPCBlogGUI, ilPCQuestionOverviewGUI, ilPCSkillsGUI
  * @ilCtrl_Calls ilPageEditorGUI: ilPCConsultationHoursGUI, ilPCMyCoursesGUI, ilPCAMDPageListGUI
  * @ilCtrl_Calls ilPageEditorGUI: ilPCGridGUI, ilPCGridCellGUI, ilPageEditorServerAdapterGUI
+ * @ilCtrl_Calls ilPageEditorGUI: ilPCCurriculumGUI, ilPCLauncherGUI
```
Comment (alex40724) [inline Services/COPage/classes/class.ilPageEditorGUI.php]: It should be possible to move this to the ilPCCurriculumGUI, ilPCLauncherGUI classes and use a "isCalledBy" comment.

File: xml/ilias_pg_5_4.dtd [hunk 2]
```diff
@@ -23,7 +23,7 @@
 <!-- PageObjects are comparable to LOM LearningObjects of aggregation level 2 -->
 <!ELEMENT PageObject (PageContent*, Layout?, InitOpenedContent?)>
 
-<!ELEMENT PageContent ((Paragraph | MediaObject | Table | List | FileList | Question | PlaceHolder | Section | Resources | LoginPageElement | Map | Plugged | Tabs | ContentInclude | InteractiveImage | Profile | Verification | Blog | QuestionOverview | Skills | LearningHistory | ConsultationHours | MyCourses | AMDPageList | AMDForm | Grid), BibItemIdentifier*)>
+<!ELEMENT PageContent ((Paragraph | MediaObject | Table | List | FileList | Question | PlaceHolder | Section | Resources | LoginPageElement | Map | Plugged | Tabs | ContentInclude | InteractiveImage | Profile | Verification | Blog | QuestionOverview | Skills | LearningHistory | ConsultationHours | MyCourses | AMDPageList | AMDForm | Grid | Curriculum | Launcher), BibItemIdentifier*)>
```
Comment (alex40724) [inline xml/ilias_pg_5_4.dtd]: Current version used in the trunk is ilias_pg_8.dtd.

**Review by alex40724 — COMMENTED** [review]
Thanks for the PR. Two changes are necessary (see comments)

- [x] ILIAS 8 uses ilias_pg_8.dtd
- [x] The ctrl comments should be moved to the PC classes.

I know it is still a mess to add new PC-Components. But I am glad that the remaining changes necessary in COPage are related to the XSL handling. We need a generic way to integrate here. I will add this to the Roadmap.

**Review by alex40724 — CHANGES_REQUESTED** [review]
See last comment.

**Comment by klees** [comment]
Thx @alex40724, it's done.

**Comment by nhaagen** [comment]
@catenglaender added an placeholder-icon. Thank you

**Pull request merged**